### PR TITLE
ConnectToTargetDialog waits for the process

### PR DESCRIPTION
**Pull request by reichlfl**
The ConnectToTargetDialog now uses the ProcessManager to wait for the
process to be found, and only returns once this is the case. I deliverately
did not add a timeout because the user can cancel this at any time.

Bug: b/218502384
Test: Manual testing so far - works when the process is already running as
well as when the process is starting while the dialog is already open, and
tested aborting in multiple states. Will add a happy path E2E test next.

### CODE REVIEW

File: src/SessionSetup/ConnectToTargetDialog.cpp [hunk 1]
```diff
@@ -75,46 +66,66 @@ std::optional<TargetConfiguration> ConnectToTargetDialog::Exec() {
...
+  stadia_connection_ = orbit_session_setup::StadiaConnection(
+      std::move(instance), std::move(service_deploy_manager), std::move(grpc_channel));
+
+  process_manager_ = orbit_client_services::ProcessManager::Create(
+      stadia_connection_.value().GetGrpcChannel(), absl::Milliseconds(1000));
+  process_manager_->SetProcessListUpdateListener(
+      [this](std::vector<orbit_grpc_protos::ProcessInfo> process_list) {
+        OnProcessListUpdate(std::move(process_list));
+      });
+  SetStatusMessage("Waiting for process to launch.");
```
Comment (antonrohr) [inline src/SessionSetup/ConnectToTargetDialog.cpp:114]: pers pref: Display `target_.process_name_or_path` in this message

**Review by antonrohr — APPROVED** [review]
lgtm

File: src/SessionSetup/ConnectToTargetDialog.cpp (hunk 1 above)
Comment (reichlfl) [reply src/SessionSetup/ConnectToTargetDialog.cpp:114]: There's a label right above this message that displays the process name, so I think this is not needed here.

Comment (antonrohr) [reply src/SessionSetup/ConnectToTargetDialog.cpp:114]: Did not realize this. Then please ignore

**Pull request merged**